pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -684,24 +684,5 @@ func (r *repository) deserializedManifestFromImage(image *imageapi.Image) (*sche
 }
 
 func (r *repository) checkPendingErrors(ctx context.Context) error {
-	return checkPendingErrors(context.GetLogger(r.ctx), ctx, r.namespace, r.name)
-}
-
-func checkPendingErrors(logger context.Logger, ctx context.Context, namespace, name string) error {
-	if !AuthPerformed(ctx) {
-		return fmt.Errorf("openshift.auth.completed missing from context")
-	}
-
-	deferredErrors, haveDeferredErrors := DeferredErrorsFrom(ctx)
-	if !haveDeferredErrors {
-		return nil
-	}
-
-	repoErr, haveRepoErr := deferredErrors.Get(namespace, name)
-	if !haveRepoErr {
-		return nil
-	}
-
-	logger.Debugf("Origin auth: found deferred error for %s/%s: %v", namespace, name, repoErr)
-	return repoErr
+	return checkPendingErrors(ctx, r.namespace, r.name)
```

Comment:
> what is the difference between the ctx passed in here and the ctx held by the repo. it was being done this way to preserve the original logging, but I don't know what additional info is logged by the ctx held by the repo

In general, the most up to date context should be used for logging. Theoretically, context held by the repo may be stale, which would result in incorrect logging statements. But most probably, the two context objects will be the same.

Using `r.ctx` is mostly a hold-over from times when middleware methods used to lack the context argument.